src/agroservices/ipm.py

```diff
@@ -137,131 +155,336 @@ def post_schema_weatherdata_validate(self):
+        weatherStationId=101104
+        ):
+        """
+        Get weather Observation for the different WeatherAdapterService available on ipm decision project
+        
+        Parameters:
         -----------
-        ignoreErrors: (Bolean) Set to "true" if you want the service to return weather data regardless of there being errors in the service
-        interval: (int) he measuring interval in seconds. Please note that the only allowed interval in this version is 3600 (hourly)
-        parameters: (list)  Comma separated list of the requested weather parameters
-        timeStart: Start of weather data period (ISO-8601 Timestamp, e.g. 2020-06-12T00:00:00+03:00)
-        timeEnd: End of weather data period (ISO-8601 Timestamp, e.g. 2020-07-03T00:00:00+03:00)
-        weatherStationId: The weather station id (FMISID) in the open data API https://en.ilmatieteenlaitos.fi/observation-stations?filterKey=groups&filterQuery=weather
+            endpoint: the endpoint corresponding to one weatheradapterservice except forecast endpoint
+                      (the list of available endpoints can be consulted using list_weatheradapter_service function)
+            credentials: (depend of the weatheradapterservice) json object with "userName" and "password" properties set 
+                         (eg: {"userName":"XXXXX","password":"XXXX"})
+            ignoreErrors: (Bolean) Set to "true" if you want the service to return weather data regardless of there being errors in the service
+            interval: (int) he measuring interval in seconds. Please note that the only allowed interval in this version is 3600 (hourly)
+            parameters: (list)  Comma separated list of the requested weather parameters
+            timeStart: Start of weather data period (ISO-8601 Timestamp, e.g. 2020-06-12T00:00:00+03:00)
+            timeEnd: End of weather data period (ISO-8601 Timestamp, e.g. 2020-07-03T00:00:00+03:00)
+            weatherStationId: The weather station id (FMISID) 
+                              in the open data API https://en.ilmatieteenlaitos.fi/observation-stations?filterKey=groups&filterQuery=weather
 
         Returns:
         --------
-         weather observations in the IPM Decision's weather data format from the Finnish Meteorological Institute https://en.ilmatieteenlaitos.fi/ in json format
+            weather observations in the IPM Decision's weather data format in json format
         """
-        pass
+        # test endpoint argument
+        endpoints = self.weatheradapter_service(forecast=False)
+        if not endpoint in endpoints.values():
+            raise ValueError("endpoint error: weatheradapter service not exit \n"
+                             "or is a forecast weatheradapter in this case used weatheradapter_forecast")
+        
+        # params according to weather adapterservice (endpoints), difference if or not credentials
+        if (credentials is not None or not credentials):
```

Comment:
Do not understand this test!
The negation is if credential is None *AND* credential, which is always False.
So this test is ALWAYS True.
Seems like a bug...

Reply:
Basically I wanted to say if credentials is not None or empty: you take into account the credentials parameter. 
To simplify, I set by default credentials is None and if it's None you don't take into account the credentials.